bookkeeper-server/src/main/java/org/apache/bookkeeper/sasl/SaslClientState.java

```diff
@@ -50,7 +50,9 @@
     private String password;
 
     public SaslClientState(String serverHostname, Subject subject) throws SaslException {
-        String serverPrincipal = SaslConstants.SASL_BOOKKEEPER_PROTOCOL + "/" + serverHostname;
+        String saslServiceName = System.getProperty("bookkeeper.sasl.servicename",
```

Comment:
This should be configurable in Client Configuration and in ServerConfiguration.

We should add tests as well, you can check existing tests and start from them

Reply:
Sorry I couldn't find any existing tests that use SaslClientState.

And this only affect client side, when client authenticate with server, you need to provide server's service name.

Reply:
@eolivelli, We will also add some other JVM properties for sasl, using system property is fine for me.  Comparing with add in `Client Configuration`, this change made less changes. 
@29x10 FYI. `GSSAPIBookKeeperTest` is the test case.

Reply:
@jiazhai I just add the test case, please help to review